contracts/test/v0.8/dev/ScrollCrossDomainForwarder.test.ts

```diff
@@ -0,0 +1,259 @@
+import { ethers } from 'hardhat'
+import { assert, expect } from 'chai'
+import { Contract, ContractFactory } from 'ethers'
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers'
+import { publicAbi } from '../../test-helpers/helpers'
+
+let owner: SignerWithAddress
+let stranger: SignerWithAddress
+let l1OwnerAddress: string
+let newL1OwnerAddress: string
+let forwarderFactory: ContractFactory
+let greeterFactory: ContractFactory
+let crossDomainMessengerFactory: ContractFactory
+let crossDomainMessenger: Contract
+let forwarder: Contract
+let greeter: Contract
+
+before(async () => {
```

Comment:
Please consider Foundry tests. We've switched to Foundry a year ago

Reply:
Apologies about that - my understanding was that hardhat was still being used

I can follow up on this task in an additional PR